examples/step-71/step-71.cc

```diff
@@ -0,0 +1,3765 @@
+    // and $\mathbf{F}$ is the deformation gradient tensor.
+    // To give some brief background to the various components of $\psi_{0}$, the first
+    // two terms bear a great resemblance to the stored energy density function for a
+    // (hyperelastic) Neohookean material. The only difference between what's used
+    // here and the Neohookean material is the scaling of the elastic shear modulus
+    // by the magnetic field-sensitive saturation function 
+    // $f_{\mu_{e}} \left( \boldsymbol{\mathbb{H}} \right)$ (@citep Pelteret2018a equation 29). This function will,
+    // in effect, cause the material to stiffen in the presence of a strong magnetic
+    // field. As its governed by a sigmoid-type function, the shear modulus will
+    // asymptotically converge on the specified saturation shear modulus.
+    // It can also be shown that the last term in $\psi_{0}$ is the 
+    // stored energy density function for magnetic field (as derived from first principles), scaled by the relative
+    // permeability constant. This definition collectively implies that the material
+    // is linearly magnetized, i.e. the magnetization vector and magnetic field
+    // vector are aligned.
+    //
+    // Since we expect that this class fully describes a single material, we'll
+    // mark it as "final" so that the inheritance tree terminated here.
+    template <int dim, AD::NumberTypes ADTypeCode>
+    class Magnetoelastic_Constitutive_Law_AD final
+      : public Coupled_Magnetomechanical_Constitutive_Law_Base<dim>
+    {
+      // Here we fefine the helper type that we will use in the AD computations for our
+      // scalar energy density function. Note that we expect it to return values of
+      // type double.
+      using ADHelper     = AD::ScalarFunction<dim, ADTypeCode, double>;
+      using ADNumberType = typename ADHelper::ad_type;
+
+    public:
+      Magnetoelastic_Constitutive_Law_AD(
+        const ConstitutiveParameters &constitutive_parameters);
+
+      // Since the public interface to the base class is pure-virtual, here
+      // we'll declare that this class will override all of these base class
+      // methods. 
+      void update_internal_data(const Tensor<1, dim> &         H,
+                                const SymmetricTensor<2, dim> &C,
+                                const DiscreteTime &) override;
+
+      double get_psi() const override;
+
+      Tensor<1, dim> get_B() const override;
+
+      SymmetricTensor<2, dim> get_S() const override;
+
+      SymmetricTensor<2, dim> get_DD() const override;
+
+      Tensor<3, dim> get_PP() const override;
+
+      SymmetricTensor<4, dim> get_HH() const override;
+
+    private:
+      // We need to define some extractors that will help us set independent variables
+      // and later get the computed values related to the dependent
+      // variables. If this were class were to be used in the context of a finite
```

Comment:
There is a problem with this sentence.

Reply:
Note to self: Perhaps I wanted to say `If this were class were to...` -> `If this class were to...`